mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -780,7 +718,78 @@ impl S3CrtClientInner {
             parse_meta_request_error,
             move |result| _ = tx.send(result.map(|_| std::mem::take(&mut *body.lock().unwrap()))),
```

Comment:
I know this isn't changing, but is panicking really the best option here compared to say just failing this specific request with an internal error?

Reply:
`unwrap` here is just for the lock, it will fail if another thread has panicked while holding the lock